runtime/src/main/java/org/corfudb/runtime/collections/TableMetrics.java

```diff
@@ -0,0 +1,224 @@
+package org.corfudb.runtime.collections;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import com.google.gson.JsonObject;
+import org.corfudb.common.metrics.micrometer.MeterRegistryProvider;
+
+import java.util.Optional;
+
+
+/**
+ * List of metrics captured at a CorfuStore Table level
+ *
+ * Created by hisundar on 2020-09-22
+ *
+ */
+public class TableMetrics {
+    private final String tableName;
+    private final Optional<Counter> numPuts;
+    private final Optional<Counter> numEnqueues;
+    private final Optional<Counter> numMerges;
+    private final Optional<Counter> numTouches;
+    private final Optional<Counter> numDeletes;
+    private final Optional<Counter> numClears;
+    private final Optional<Counter> numGets;
+    private final Optional<Counter> numCounts;
+    private final Optional<Counter> numKeySets;
+    private final Optional<Counter> numScans;
+    private final Optional<Counter> numEntryLists;
+    private final Optional<Counter> numJoins;
+    private final Optional<Counter> numGetByIndex;
+    private final Optional<Counter> numTxnAborts;
+
+    private final TimingHistogram writeOnlyTxnTimes;
+    private final TimingHistogram readOnlyTxnTimes;
+    private final TimingHistogram readWriteTxnTimes;
+
+    public void incNumPuts() {
+        if (!numPuts.isPresent()) {
+            return;
+        }
+        numPuts.get().increment();
+    }
```

Comment:
`numPuts.ifPresent(Counter::increment);` seems to be more concise.

Reply:
done, thanks!